include/zenoh-pico/collections/pointer.h

```diff
@@ -30,9 +30,15 @@
     static inline name##_sptr_t name##_sptr_new(type##_t val) {                                 \
         name##_sptr_t p;                                                                        \
         p.ptr = (type##_t *)z_malloc(sizeof(type##_t));                                         \
-        *p.ptr = val;                                                                           \
         p._cnt = (atomic_uint *)z_malloc(sizeof(atomic_uint));                                  \
```

Comment:
This malloc should be inside the `if (p.ptr != NULL)`. If for any obscure reason the first malloc fails and the second one succeeds, we have a memory leak because the corresponding cleanup is not done.

Reply:
Done in the upcoming commits.